draft-group-privacypass-consistency-mirror.md

```diff
@@ -313,6 +313,34 @@ be preferable to protocol-agnostic consistency checks based on the mirror protoc
 especially if they provide equivalent consistency guarantees with better performance
 or reliability.
 
+## Consistency Validity Period
+
+Executing a consistency check provides a client with some assurance that other
+clients may be using the same resource {{security-considerations}}. However,
+this result only reflects a mirror's view of a resource at a particular point
+in time. The client should periodically re-check consistency. The frequency of
+re-checking depends on the requirements of the client {{integration}}. Clients
+should re-confirm consistency of a cached resource if it is not fresh (see
+{{Section 4.2 of ?CACHING=RFC9111}}).
+
+Two strategies for maintaining consistency are:
+1. Pre-emptively execute a consistency check for a resource that is expiring soon; and
+1. Execute a consistency check of an expired resource at the time of its next use.
+
+For strategy 1, clients that execute consistency checks at the same time can
+induce a thundering herd that overwhelms the mirror resource. Since coordinating
+consistency checks across clients is difficult, clients can instead execute consistency
+checks at random times before the resource expires. Clients that have more information
```

Comment:
I don't think this helps.  The resource is only valid until its expiration time, and repeated checking doesn't extend that time.  Once the resource expires, all clients will have to refresh it before they can rely on it again.

Reply:
But that's where the following come into play:
```
When the origin server has multiple versions of a resource corresponding to a
 URL, it should respond with the resource that is both currently valid and will
 remain fresh for the longest amount of time in the future.
```

In other words, if an origin is correctly configured and has either 1) a new resource that it is rotating to in the (near) future when the current resource expires, or 2) an extended expiration time/period for the current resource, then it should respond with the freshest and long-living answer it has available. Ideally, this will happen at least a few hours before the previous expiration time - if not longer.

Perhaps, saying this more explicitly, the service should be updated with a new expiration time or resource well before the current expiration time is reached. Do you believe this isn't a reasonable expectation?

Reply:
I don't think this works.  The mirror will serve all responses out of cache until its cached copy expires.  It won't contact the origin server at all before then, so "early rotation" by the origin server has no effect.  (The mirror cannot simply update its cached copy early, as this partitions the clients into epochs.)

More comments here: https://github.com/chris-wood/draft-group-privacypass-consistency-mirror/issues/8#issuecomment-1775617206